pandas/core/indexes/numeric.py

```diff
@@ -381,7 +381,7 @@ def get_loc(self, key, method=None, tolerance=None):
             # Catch this to avoid accidentally casting to 1.0
             raise KeyError(key)
 
-        if is_float(key) and np.isnan(key):
+        if is_float(key) and np.isnan(key) and method is None:
```

Comment:
in the case with method not-None, can we infer that the super call is going to raise?

BTW im pretty sure we could remove this method altogether if we got the comparison in index.pyx to be nan-safe

Reply:
I look into the removing part.

Would returning NaN be more sensible in this case?

Reply:
Something like this? We have to keep the is_bool check, to avoid casting to 1.0 or 0.0